### ISSUE TITLE
accept es6 iterators for data?

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1567]
Aside from our use of d3 selections, which will accept iterables d3/d3-selection#209, I doubt there are too many places where we rely on `group.all()` to return an array rather than a sequence.

When d3-selection 2.0 lands, we should see if we can accept iterators. Probably there would be a long tail of bugs on this, but hopefully the test suite will cover most of them.

**Comment by kum-deepak** [comment]
Can this be closed?

**Comment by gordonwoodhull** [comment]
I don’t think so. My understanding is that it does take some changes to accommodate iterators.
